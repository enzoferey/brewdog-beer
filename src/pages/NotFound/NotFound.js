import React from "react";
import PropTypes from "prop-types";
import style from "./not-found.scss";

import { Link } from "react-router-dom";

const NotFound = ({ message }) => (
  <div className={style.main}>
    <h2>{message}</h2>
    <Link to="/">Go to Home</Link>
  </div>
);

NotFound.propTypes = {
  message: PropTypes.string,
};
NotFound.defaultProps = {
  message: "Woops, this page doesn't seem to exist",
};

export default NotFound;
