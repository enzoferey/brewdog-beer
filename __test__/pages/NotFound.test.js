import React from "react";
import NotFound from "pages/NotFound";

const NotFoundRouter = withRouter(NotFound);

test("renders correctly", () => {
  testSnapshot(<NotFoundRouter />);
});

test("renders correctly `message` prop", () => {
  testSnapshot(<NotFoundRouter message="hello" />);
});
