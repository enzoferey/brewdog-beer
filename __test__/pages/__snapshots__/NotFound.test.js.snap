// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly \`message\` prop 1`] = `
<div>
  <h2>
    hello
  </h2>
  <a
    href="/"
    onClick={[Function]}
  >
    Go to Home
  </a>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <h2>
    Woops, this page doesn't seem to exist
  </h2>
  <a
    href="/"
    onClick={[Function]}
  >
    Go to Home
  </a>
</div>
`;
